Tools/clinic/clinic.py

```diff
@@ -1250,18 +1250,32 @@ def parser_body(
             parser_prototype = self.PARSER_PROTOTYPE_VARARGS
             parser_definition = parser_body(parser_prototype, '    {option_group_parsing}')
 
-        elif not requires_defining_class and pos_only == len(parameters) - pseudo_args and clinic.limited_capi:
-            # positional-only for the limited C API
-            flags = "METH_VARARGS"
+        elif clinic.limited_capi:
+            if not requires_defining_class and pos_only == len(parameters) - pseudo_args:
+                # positional-only for the limited C API:
+                # PyArg_ParseTuple()
+                flags = "METH_VARARGS"
 
-            parser_prototype = self.PARSER_PROTOTYPE_VARARGS
-            parser_code = [normalize_snippet("""
-                if (!PyArg_ParseTuple(args, "{format_units}:{name}",
-                    {parse_arguments}))
-                    goto exit;
-            """, indent=4)]
-            argname_fmt = 'args[%d]'
-            declarations = ""
+                parser_prototype = self.PARSER_PROTOTYPE_VARARGS
+                parser_code = [normalize_snippet("""
+                    if (!PyArg_ParseTuple(args, "{format_units}:{name}",
+                        {parse_arguments}))
+                        goto exit;
+                """, indent=4)]
+                declarations = ""
+
+            else:
```

Comment:
Does it work if `requires_defining_class` is true?

Reply:
No, it doesn't. This change is incomplete, it can be completed later. Or do you mean that an exception should be raised instead of an compilation error?

Reply:
Currently it can produce invalid code which is compiled. There are two opposite approaches:

1. Raise an exception or ensure that the compiling the generated code will get a compilation error.
2. Try to use the limited API if possible, but fallback to private API otherwise.

I am currently trying the second approach. It allows to re-generate all clinic code using the limited API and test that at least it works (I found many cases in which the current code does not generate working code). Please hold this PR, the current code is more suitable for this experiment. After testing it will be easy to switch to the first approach.

Reply:
A module targeting the limited C API cannot use the private API. It fails at build.